### PR TITLE
Preserve target order defined in `Project.swift` when generating project

### PR DESCRIPTION
**Pull request by moritzsternemann**
Resolves https://github.com/tuist/tuist/pull/4707#issuecomment-1235491849

### Short description 📝

The `TreeShakePrunedTargetsGraphMapper` didn't preserve the order of targets as they were defined in Project.swift. This was caused by copying the target list from the `Graph.targets` dictionary, which doesn't preserve the order.

As discussed in [this Slack thread](https://tuistapp.slack.com/archives/CCXC295QS/p1665067784600719), it now directly uses the `Project.targets` array, which preserves the order.

### How to test the changes locally 🧐

I added a unit test with targets that would've been ordered differently when applying the `TreeShakePrunedTargetsGraphMapper` before the patch. The example names come from [Apple's documentation on the order of elements in the `Dictionary.values` collection](https://developer.apple.com/documentation/swift/dictionary/values-swift.property).

### Checklist ✅

- [x] The code architecture and patterns are consistent with the rest of the codebase
- [x] The title of the PR will be used as changelog entry, please make sure it is clear and suitable
- [ ] In case the PR introduces changes that affect users, the documentation has been updated
- [x] Contributors have checked that, if needed, the PR includes the label `changelog:added`, `changelog:fixed`, or `changelog:changed`

### CODE REVIEW

File: Tests/TuistCacheTests/Mappers/Graph/TreeShakePrunedTargetsGraphMapperTests.swift [hunk 1]
```diff
@@ -241,4 +241,36 @@ final class TreeShakePrunedTargetsGraphMapperTests: TuistUnitTestCase {
...
+            dependencies: [:]
+        )
+
+        let expectedProject = Project.test(targets: [firstTarget, secondTarget, thirdTarget])
+
+        // When
+        let (gotGraph, _) = try subject.map(graph: graph)
+
+        print(graph.projects.first!.value.targets.map(\.name))
+        print(gotGraph.projects.first!.value.targets.map(\.name))
```
Comment (danyf90) [inline Tests/TuistCacheTests/Mappers/Graph/TreeShakePrunedTargetsGraphMapperTests.swift:271]: Shouldn't this XCTAssert on it rather than printing? :D

**Review by danyf90 — CHANGES_REQUESTED** [review]

**Comment by danyf90** [comment]
Please run `./fourier lint tuist --fix` to fix linting 🙏

File: Tests/TuistCacheTests/Mappers/Graph/TreeShakePrunedTargetsGraphMapperTests.swift (hunk 1 above)
Comment (shahzadmajeed) [reply Tests/TuistCacheTests/Mappers/Graph/TreeShakePrunedTargetsGraphMapperTests.swift:271]: If we are making sure that order of targets is the same, does it make sense to zip both sequences and just compare them?

i.e, 

```
for (generatedTarget, expectedTarget) in zip(graph.projects.first!.value.targets, expectedProject.targets) {
     XCTAssertEqual(generatedTarget expectedTarget)
}
```

Comment (moritzsternemann) [reply Tests/TuistCacheTests/Mappers/Graph/TreeShakePrunedTargetsGraphMapperTests.swift:271]: Ahh I forgot them in there.

I think the `XCTAssertEqual(gotGraph.projects.first?.value, expectedProject)` is enough to cover the order of projects based on the `Equatable` conformance of `Project`. Let me know if you think the assertion should be more specific.

Comment (danyf90) [reply Tests/TuistCacheTests/Mappers/Graph/TreeShakePrunedTargetsGraphMapperTests.swift]: You are right, I didn't consider that comparing the projects we were also comparing the targets 😁

**Review by danyf90 — APPROVED** [review]

**Pull request merged**